eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -228,34 +232,19 @@ void shutdown() {
         }
     }
 
-    /**
-     * Assumes applicationInfoManager is already initialized
-     *
-     * @deprecated use constructor that takes ApplicationInfoManager instead of InstanceInfo directly
-     */
-    @Deprecated
-    public DiscoveryClient(InstanceInfo myInfo, EurekaClientConfig config) {
-        this(myInfo, config, null);
-    }
-
-    /**
-     * @deprecated use constructor that takes ApplicationInfoManager instead of InstanceInfo directly
-     */
-    @Deprecated
-    public DiscoveryClient(InstanceInfo myInfo, EurekaClientConfig config, AbstractDiscoveryClientOptionalArgs args) {
-        this(ApplicationInfoManager.getInstance(), config, args);
-    }
-
-    public DiscoveryClient(ApplicationInfoManager applicationInfoManager, EurekaClientConfig config) {
-        this(applicationInfoManager, config, null);
+    @SuppressWarnings("rawtypes")
```

Comment:
Now TransportClientFactories is mandatory. This would break anyone who doesn't use SB cloud. Is there a way that we can construct a default TransportClientFactories for users and provide ctor option w/o TransportClientFactories? I know it's a major release, but I'm not sure how users would know about what to do to construct a TransportClientFactories?

Reply:
Not without merging jersey 3 modules. Spring cloud netflix with auto-configure the right thing. Right now there is no guice support, so until that happens no one can use it without spring cloud netflix anyway. If you do get guice support, then that would do the right thing.